presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/H2TestingUtil.java

```diff
@@ -0,0 +1,174 @@
+    }
+
+    static Session newRejectionSession()
+    {
+        return testSessionBuilder()
+                .setCatalog("tpch")
+                .setSchema("sf100000")
+                .setSource("reject")
+                .build();
+    }
+
+    static void waitForCompleteQueryCount(DistributedQueryRunner queryRunner, int expectedCount)
+            throws InterruptedException
+    {
+        waitForQueryCount(queryRunner, TERMINAL_QUERY_STATES, expectedCount);
+    }
+
+    static void waitForRunningQueryCount(DistributedQueryRunner queryRunner, int expectedCount)
+            throws InterruptedException
+    {
+        waitForQueryCount(queryRunner, ImmutableSet.of(RUNNING), expectedCount);
+    }
+
+    static void waitForQueryCount(DistributedQueryRunner queryRunner, Set<QueryState> countingStates, int expectedCount)
+            throws InterruptedException
+    {
+        QueryManager queryManager = queryRunner.getCoordinator().getQueryManager();
+        while (queryManager.getAllQueryInfo().stream().filter(q -> countingStates.contains(q.getState())).count() != expectedCount) {
+            MILLISECONDS.sleep(500);
+        }
+    }
+
+    static String getDbConfigUrl()
+    {
+        Random rnd = new Random();
+        return "jdbc:h2:mem:test_" + Math.abs(rnd.nextLong());
+    }
+
+    static H2ResourceGroupsDao getDao(String url)
+    {
+        DbResourceGroupConfig dbResourceGroupConfig = new DbResourceGroupConfig()
+                .setConfigDbUrl(url);
+        H2ResourceGroupsDao dao = new H2DaoProvider(dbResourceGroupConfig).get();
+        dao.createResourceGroupsTable();
+        dao.createSelectorsTable();
+        dao.createResourceGroupsGlobalPropertiesTable();
+        return dao;
+    }
+
+    static DistributedQueryRunner createQueryRunner(String dbConfigUrl, H2ResourceGroupsDao dao)
+            throws Exception
+    {
+        ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+        builder.put("experimental.resource-groups-enabled", "true");
+        Map<String, String> properties = builder.build();
+        DistributedQueryRunner queryRunner = new DistributedQueryRunner(testSessionBuilder().build(), 2, ImmutableMap.of(), properties, new SqlParserOptions());
+        try {
+            Plugin h2ResourceGroupManagerPlugin = new H2ResourceGroupManagerPlugin();
+            queryRunner.installPlugin(h2ResourceGroupManagerPlugin);
+            queryRunner.getCoordinator().getResourceGroupManager().get()
+                    .setConfigurationManager(NAME, ImmutableMap.of("resource-groups.config-db-url", dbConfigUrl));
+            queryRunner.installPlugin(new TpchPlugin());
+            queryRunner.createCatalog("tpch", "tpch");
+            setup(queryRunner, dao);
+            return queryRunner;
+        }
+        catch (Exception e) {
+            queryRunner.close();
+            throw e;
+        }
+    }
+
+    static DistributedQueryRunner getSimpleQueryRunner()
```

Comment:
Could you explain how these two methods are different? Can common logic be factored out?